fenix/app/src/main/java/org/mozilla/fenix/addons/ExtensionProcessDisabledController.kt

```diff
@@ -84,4 +39,67 @@ class ExtensionProcessDisabledController(
 ) : ExtensionProcessDisabledPopupObserver(
     store,
     { presentDialog(context, store, engine, builder, appName) },
-)
+) {
+    override fun onDestroy(owner: LifecycleOwner) {
+        super.onDestroy(owner)
+        // In case the activity gets destroyed, we want to reset the dialog so that it gets re-created when needed.
+        dialog = null
+    }
+
+    companion object {
+        private var dialog: AlertDialog? = null
```

Comment:
Ideally we shouldn't keep Android UI components as static references, it's. common Android code smell. As we could cause memory leaks. I noticed you're cleaning up the reference on `onDestroy`, but but we will be introducing brittle pattern. Let's try to think about other alternatives. I'll try to find a possible solution and let you know!

Reply:
Have we tried this PR with "Don't keep activities" setting enabled ?

Reply:
Another alternative instead of having a reference of `AlertDialog`, we could have a `boolean` which indicates if the dialog is already showing, what do you think?

Reply:
> have you already tried other alternatives?

No

> Have we tried this PR with "Don't keep activities" setting enabled ?

Yes, hence the `onDestroy()` call 🙈 

> Another alternative instead of having a reference of AlertDialog, we could have a boolean which indicates if the dialog is already showing, what do you think?

I suppose that could work yeah, let me try.